pkg/server/batch_check.go

```diff
@@ -0,0 +1,130 @@
+	"github.com/openfga/openfga/internal/authz"
+	"github.com/openfga/openfga/pkg/server/commands"
+	"github.com/openfga/openfga/pkg/telemetry"
+	"go.opentelemetry.io/otel/attribute"
+	"go.opentelemetry.io/otel/trace"
+)
+
+func (s *Server) BatchCheck(ctx context.Context, req *openfgav1.BatchCheckRequest) (*openfgav1.BatchCheckResponse, error) {
+	ctx, span := tracer.Start(ctx, authz.BatchCheck, trace.WithAttributes(
+		attribute.KeyValue{Key: "store_id", Value: attribute.StringValue(req.GetStoreId())},
+		attribute.KeyValue{Key: "batch_size", Value: attribute.IntValue(len(req.GetChecks()))},
+		attribute.KeyValue{Key: "consistency", Value: attribute.StringValue(req.GetConsistency().String())},
+	))
+	defer span.End()
+
+	if !validator.RequestIsValidatedFromContext(ctx) {
+		if err := req.Validate(); err != nil {
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+	}
+
+	err := s.checkAuthz(ctx, req.GetStoreId(), authz.BatchCheck)
+	if err != nil {
+		return nil, err
+	}
+
+	ctx = telemetry.ContextWithRPCInfo(ctx, telemetry.RPCInfo{
+		Service: s.serviceName,
+		Method:  authz.BatchCheck,
+	})
+
+	storeID := req.GetStoreId()
+
+	typesys, err := s.resolveTypesystem(ctx, storeID, req.GetAuthorizationModelId())
+	if err != nil {
+		return nil, err
+	}
+
+	cmd := commands.NewBatchCheckCommand(
+		s.checkDatastore,
+		s.checkResolver,
+		typesys,
+		commands.WithBatchCheckCommandCacheController(s.cacheController),
+		commands.WithBatchCheckCommandLogger(s.logger),
+		commands.WithBatchCheckMaxChecksPerBatch(s.maxChecksPerBatchCheck),
+		commands.WithBatchCheckMaxConcurrentChecks(s.maxConcurrentChecksPerBatch),
+	)
+
+	result, metadata, err := cmd.Execute(ctx, &commands.BatchCheckCommandParams{
+		AuthorizationModelID: typesys.GetAuthorizationModelID(),
+		Checks:               req.GetChecks(),
+		Consistency:          req.GetConsistency(),
+		StoreID:              storeID,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	grpc_ctxtags.Extract(ctx).Set(datastoreQueryCountHistogramName, metadata.TotalQueries)
+
+	return &openfgav1.BatchCheckResponse{Result: transformCheckResultToRPC(result)}, nil
+}
+
+// transformCheckResultToRPC transform the internal BatchCheckOutcome into the external-facing
+// BatchCheckSingleResult struct for transmission back via the api.
+func transformCheckResultToRPC(checkResults map[commands.CorrelationID]*commands.BatchCheckOutcome) map[string]*openfgav1.BatchCheckSingleResult {
+	var batchResult = map[string]*openfgav1.BatchCheckSingleResult{}
+	for k, v := range checkResults {
+		singleResult := &openfgav1.BatchCheckSingleResult{}
+
+		if v.Err != nil {
+			singleResult.CheckResult = &openfgav1.BatchCheckSingleResult_Error{
+				Error: transformCheckCommandErrorToBatchCheckError(v.Err),
+			}
+		} else {
+			singleResult.CheckResult = &openfgav1.BatchCheckSingleResult_Allowed{
+				Allowed: v.CheckResponse.Allowed,
+			}
+		}
+
+		batchResult[string(k)] = singleResult
```

Comment:
How is this not covered by tests?

Reply:
There are no tests in this branch anywhere to try and make review easier. There's a separate [PR here which is all tests](https://github.com/openfga/openfga/pull/2047/files). The plan is to eventually merge that testing PR into this branch, at which point these codecov warnings should all disappear

Reply:
Update: #2065 (unit tests) and #2066 (integration tests) have been merged into this branch.